### ISSUE TITLE
Incorrect iam policy

### ISSUE THREAD

**Issue by kumud-pe** [kubernetes-sigs/aws-alb-ingress-controller#748]
The documented policy doesn't have:

"ec2:DescribeAccountAttributes"
"ec2:DescribeInternetGateways"
"iam:CreateServiceLinkedRole"

But, the controller needs them.

**Comment by M00nF1sh** [comment]
Hi, i don't think the controller need these permissions. Would you help paste the relevant logs for this 😄

**Comment by kumud-pe** [comment]
Looks like somehow this is tied to #749 (or may be not). I am not getting these error logs anymore.
BTW. I am using kube2iam for granting AWS access to the controller.
I got error messages intially when I installed controller for the first time and adding these fixed the issue.

**Comment by vincentheet** [comment]
When deploying an ingress type we had similar issues as stated above. The aws-alb-ingress-controller was not able to create a LoadBalancer because of permission issues as can be seen in the attached logs. After adding the permissions "ec2:DescribeAccountAttributes", "ec2:DescribeInternetGateways", "iam:CreateServiceLinkedRole" the issue was resolved.

When looking into the logs we saw that the api call `elasticloadbalancing/CreateLoadBalancer` fails due to this permission error `is not authorized to perform: ec2:DescribeAccountAttributes`

The AWS ELB API permissions documentation states that for `CreateLoadBalancer` the permissions are needed: https://docs.aws.amazon.com/elasticloadbalancing/latest/userguide/elb-api-permissions.html

```I1129 14:08:12.704972       1 api.go:159] Request: tagging/GetResources, Payload: {  ResourceTypeFilters: ["ec2"],  ResourcesPerPage: 50,  TagFilters: [{      Key: "kubernetes.io/role/internal-elb",      Values: ["","1"]    },{      Key: "kubernetes.io/cluster/acme-app-acceptance",      Values: ["owned","shared"]    }]}
I1129 14:08:12.705019       1 request_pagination.go:105] Response: tagging/GetResources, Body: {  PaginationToken: "",  ResourceTagMappingList: []}
I1129 14:08:12.705184       1 api.go:159] Request: tagging/GetResources, Payload: {  ResourceTypeFilters: ["ec2"],  ResourcesPerPage: 50,  TagFilters: [{      Key: "kubernetes.io/role/elb",      Values: ["","1"]    },{      Key: "kubernetes.io/cluster/acme-app-acceptance",      Values: ["owned","shared"]    }]}
I1129 14:08:12.705449       1 request_pagination.go:105] Response: tagging/GetResources, Body: {  PaginationToken: "",  ResourceTagMappingList: [{      ResourceARN: "arn:aws:ec2:eu-west-1:511111111111:subnet/subnet-02f111111111111ab",      Tags: [        {          Key: "ApplicationName",          Value: "acme Landing Zone"        },        {          Key: "Name",          Value: "acme-public-2"        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet2"        },        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        }      ]    },{      ResourceARN: "arn:aws:ec2:eu-west-1:511111111111:subnet/subnet-0711111111111133f",      Tags: [        {          Key: "ApplicationName",          Value: "acme Landing Zone"        },        {          Key: "Name",          Value: "acme-public-1"        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet1"        },        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        }      ]    },{      ResourceARN: "arn:aws:ec2:eu-west-1:511111111111:subnet/subnet-05111111111111249",      Tags: [        {          Key: "ApplicationName",          Value: "acme Landing Zone"        },        {          Key: "Name",          Value: "acme-public-3"        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet3"        },        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        }      ]    }]}
I1129 14:08:12.705607       1 ec2.go:120] Request: ec2/DescribeSubnets, Payload: {  Filters: [{      Name: "subnet-id",      Values: ["subnet-02f111111111111ab","subnet-0711111111111133f","subnet-05111111111111249"]    },{      Name: "vpc-id",      Values: ["vpc-0251e0867ead98cca"]    }]}
I1129 14:08:12.706614       1 api.go:14070] Response: ec2/DescribeSubnets, Body: {  Subnets: [{      AssignIpv6AddressOnCreation: false,      AvailabilityZone: "eu-west-1b",      AvailableIpAddressCount: 250,      CidrBlock: "10.1.49.0/24",      DefaultForAz: false,      Ipv6CidrBlockAssociationSet: [{          AssociationId: "subnet-cidr-assoc-0ef4870e03559ee4d",          Ipv6CidrBlock: "2a05:d018:fb3:f01::/64",          Ipv6CidrBlockState: {            State: "associated"          }        }],      MapPublicIpOnLaunch: true,      State: "available",      SubnetId: "subnet-02f111111111111ab",      Tags: [        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        },        {          Key: "Name",          Value: "acme-public-2"        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet2"        },        {          Key: "ApplicationName",          Value: "acme Landing Zone"        },        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        }      ],      VpcId: "vpc-0251e0867ead98cca"    },{      AssignIpv6AddressOnCreation: false,      AvailabilityZone: "eu-west-1a",      AvailableIpAddressCount: 250,      CidrBlock: "10.1.48.0/24",      DefaultForAz: false,      Ipv6CidrBlockAssociationSet: [{          AssociationId: "subnet-cidr-assoc-0305de71411f1a3b9",          Ipv6CidrBlock: "2a05:d018:fb3:f00::/64",          Ipv6CidrBlockState: {            State: "associated"          }        }],      MapPublicIpOnLaunch: true,      State: "available",      SubnetId: "subnet-0711111111111133f",      Tags: [        {          Key: "ApplicationName",          Value: "acme Landing Zone"        },        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        },        {          Key: "Name",          Value: "acme-public-1"        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet1"        },        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        }      ],      VpcId: "vpc-0251e0867ead98cca"    },{      AssignIpv6AddressOnCreation: false,      AvailabilityZone: "eu-west-1c",      AvailableIpAddressCount: 250,      CidrBlock: "10.1.50.0/24",      DefaultForAz: false,      Ipv6CidrBlockAssociationSet: [{          AssociationId: "subnet-cidr-assoc-0aa9fbd885283c1e4",          Ipv6CidrBlock: "2a01:a017:f02:f02::/64",          Ipv6CidrBlockState: {            State: "associated"          }        }],      MapPublicIpOnLaunch: true,      State: "available",      SubnetId: "subnet-05111111111111249",      Tags: [        {          Key: "aws:cloudformation:stack-id",          Value: "arn:aws:cloudformation:eu-west-1:511111111111:stack/StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f/94c38870-de81-11e8-bed9-500c423d20d2"        },        {          Key: "aws:cloudformation:logical-id",          Value: "PublicSubnet3"        },        {          Key: "kubernetes.io/role/elb",          Value: ""        },        {          Key: "aws:cloudformation:stack-name",          Value: "StackSet-acme-vpc-f4fe14aa-6533-4b70-94f2-f80a80d9a97f"        },        {          Key: "Name",          Value: "acme-public-3"        },        {          Key: "kubernetes.io/cluster/acme-app-acceptance",          Value: "shared"        },        {          Key: "ApplicationName",          Value: "acme Landing Zone"        }      ],      VpcId: "vpc-0251e0867ead98cca"    }]}
I1129 14:08:12.706742       1 api.go:1647] Request: elasticloadbalancing/DescribeLoadBalancers, Payload: {  Names: ["79667416-default-appf-3c87"]}
E1129 14:08:12.881101       1 request_pagination.go:105] Failed request: elasticloadbalancing/DescribeLoadBalancers, Payload: {  Names: ["79667416-default-appf-3c87"]}, Error: LoadBalancerNotFound: Load balancers '[79667416-default-appf-3c87]' not found
	status code: 400, request id: 355b2638-f3e0-11e8-a63e-2723bdb3dba5
I1129 14:08:12.881125       1 loadbalancer.go:185] default/app-frontend: creating LoadBalancer 79667416-default-appf-3c87
I1129 14:08:12.881420       1 elbv2.go:118] Request: elasticloadbalancing/CreateLoadBalancer, Payload: {  IpAddressType: "ipv4",  Name: "79667416-default-appf-3c87",  Scheme: "internet-facing",  Subnets: ["subnet-02f111111111111ab","subnet-05111111111111249","subnet-0711111111111133f"],  Tags: [{      Key: "kubernetes.io/namespace",      Value: "default"    },{      Key: "kubernetes.io/ingress-name",      Value: "app-frontend"    },{      Key: "kubernetes.io/cluster/acme-app-acceptance",      Value: "owned"    }],  Type: "application"}
E1129 14:08:13.004409       1 api.go:470] Failed request: elasticloadbalancing/CreateLoadBalancer, Payload: {  IpAddressType: "ipv4",  Name: "79667416-default-appf-3c87",  Scheme: "internet-facing",  Subnets: ["subnet-02f111111111111ab","subnet-05111111111111249","subnet-0711111111111133f"],  Tags: [{      Key: "kubernetes.io/namespace",      Value: "default"    },{      Key: "kubernetes.io/ingress-name",      Value: "app-frontend"    },{      Key: "kubernetes.io/cluster/acme-app-acceptance",      Value: "owned"    }],  Type: "application"}, Error: AccessDenied: User: arn:aws:iam::511111111111:user/aws-alb-ingress-controlle-AwsAlbIngressControllerU-MM5SNS9RWAPC is not authorized to perform: ec2:DescribeAccountAttributes
	status code: 403, request id: 35681f49-f3e0-11e8-aa7e-6dbf6ed1d9d4
E1129 14:08:13.004516       1 loadbalancer.go:195] default/app-frontend: failed to create LoadBalancer 79667416-default-appf-3c87 due to AccessDenied: User: arn:aws:iam::511111111111:user/aws-alb-ingress-controlle-AwsAlbIngressControllerU-MM5SNS9RWAPC is not authorized to perform: ec2:DescribeAccountAttributes
E1129 14:08:13.004535       1 loadbalancer.go:195] default/app-frontend: 	status code: 403, request id: 35681f49-f3e0-11e8-aa7e-6dbf6ed1d9d4
E1129 14:08:13.004614       1 :0] kubebuilder/controller "msg"="Reconciler error" "error"="failed to create LoadBalancer due to AccessDenied: User: arn:aws:iam::511111111111:user/aws-alb-ingress-controlle-AwsAlbIngressControllerU-MM5SNS9RWAPC is not authorized to perform: ec2:DescribeAccountAttributes\n\tstatus code: 403, request id: 35681f49-f3e0-11e8-aa7e-6dbf6ed1d9d4"  "Controller"="alb-ingress-controller" "Request"={"Namespace":"default","Name":"app-frontend"}
```
